### PR TITLE
[LJI-42] Implement plan limit calculations

### PR DESCRIPTION
**Pull request by anddevdev**
Implemented plan limit calculation. Added totalLimit and remainingLimit to plan entity. Remaining limit will be calculated if a claim with status approved is being added, or if existing claim's status is being changed to approved. If the limit is exceeded, a message will be shown.

### CODE REVIEW

File: src/main/java/com/coherentsolutions/pot/insurance/dto/ClaimDTO.java [hunk 1]
```diff
@@ -31,9 +31,12 @@ public class ClaimDTO implements Serializable {
   @NotNull(message = "Date of service is required")
   private LocalDate dateOfService;
 
-  @NotNull(message = "Plan is required")
+  @NotNull(message = "Plan type is required")
   private ClaimPlan plan;
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/dto/ClaimDTO.java:35]: would it be better to name it "planType"?

File: src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java [hunk 2]
```diff
@@ -39,6 +39,10 @@ public class ClaimEntity extends BaseEntity {
   @Enumerated(EnumType.STRING)
   private ClaimPlan plan;
 
+  @ManyToOne
+  @JoinColumn(name = "plan_id", referencedColumnName = "id")
+  private PlanEntity planId;
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java:44]: misleading name

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: src/main/java/com/coherentsolutions/pot/insurance/dto/ClaimDTO.java (hunk 1 above)
Comment (anddevdev) [reply src/main/java/com/coherentsolutions/pot/insurance/dto/ClaimDTO.java]: Done

File: src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java (hunk 2 above)
Comment (anddevdev) [reply src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java:44]: Done, changed plan to planType

Comment (SavaLX) [reply src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java:44]:
```suggestion
  private PlanEntity planEntity;
```

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java (hunk 2 above)
Comment (anddevdev) [reply src/main/java/com/coherentsolutions/pot/insurance/entity/ClaimEntity.java]: Done

**Review by NormantasN — APPROVED** [review]

**Review by SavaLX — APPROVED** [review]

**Pull request merged**